src/frontends/onnx/frontend/src/detail/subgraph_extraction.cpp

```diff
@@ -283,17 +283,59 @@ SubgraphExtractor::SubgraphExtractor(ONNX_NAMESPACE::GraphProto& graph)
     }
 }
 
-void SubgraphExtractor::add_new_inputs(const std::vector<InputEdge>& new_inputs) {
-    for (const auto& input_edge : new_inputs) {
-        const auto tensor_name = get_input_tensor_name(m_onnx_graph, input_edge);
+void SubgraphExtractor::add_new_inputs(const std::vector<InputEdge>& new_inputs, const bool merge_inputs) {
```

Comment:
LGTM, just a nit: I'd rather prefer enum (or even tag dispatcher) instead of bool to control flow

Reply:
old C style habits, enum looks like a better solution.